utils/db/setupRepos.ts

```diff
@@ -0,0 +1,166 @@
+        console.error(`[handleReposetupAndRemoveExtra] Error in removeExtraRepositories: ${repo_provider}/${repo_owner} for install_id ${install_id}`, error);
+        throw new Error('Failed to remove extra repositories');
+    });
+};
+
+// Function to handle setup for a single repository
+const handleRepoSetup = async (repo_name: string, repo_owner: string, repo_provider: string, install_id: string) => {
+    await conn.query('BEGIN').catch(err => {
+        console.error(`[handleRepoSetup] Could not start db transaction: ${repo_provider}/${repo_owner} for install_id ${install_id}`, { pg_query: query }, err);
+        throw new Error('Failed to begin transaction');
+    });
+
+    const query = `SELECT install_id FROM repos WHERE repo_name = $1 AND repo_owner = $2 AND repo_provider = $3`;
+    const existingInstallationsResult = await conn.query(query, [repo_name, repo_owner, repo_provider]).catch(err => {
+        console.error(`[handleRepoSetup] Could not get the install-id for repository: ${repo_provider}/${repo_owner}/${repo_name} for install_id ${install_id}`, { pg_query: query }, err);
+        throw new Error("Error in getting install-id from db");
+    });
+
+    const existingInstallations = existingInstallationsResult.rows.map(row => row.install_id);
+    console.log("[handleRepoSetup] existingInstallations: ", existingInstallations)
+
+    // Iterate over each row
+    if (existingInstallations && existingInstallations.length > 0) {
+        for (const installations of existingInstallations) {
+            // Check if the install_id is already present in the array
+            if (!installations.includes(install_id)) {
+                installations.push(install_id);
+                const updateQuery = 'UPDATE repos SET install_id = $1 WHERE repo_name = $2 AND repo_owner = $3 AND repo_provider = $4';
+                await conn.query(updateQuery, [installations, repo_name, repo_owner, repo_provider])
+                    .catch(err => {
+                        console.error(`[handleRepoSetup] Could not update the install-id array for repository: ${repo_provider}/${repo_owner}/${repo_name} for install_id ${install_id}`, { pg_query: query }, err);
+                        throw new Error('Failed to update install-id array for repo');
+                    });
+            }
+        }
+    } else {
+        //If no installation exist that means add that repo into repos table with the upcoming install_id
+        const query = `INSERT INTO repos (repo_name, repo_owner, repo_provider, install_id) VALUES ($1, $2, $3, $4)`
+        await conn.query(query, [repo_name, repo_owner, repo_provider, [install_id]]).catch(err => {
+            console.error(`[handleRepoSetup] could not insert repo into repos table: ${repo_provider}/${repo_owner}/${repo_name} for install_id ${install_id}`, { pg_query: query }, err);
+            throw new Error("Error in inserting repo in db");
+        })
```

Comment:
Can be reduced to a single query which performs set union in update and normal insert